File: examples/calculator.c
/*
 * Example C program demonstrating Mistode obfuscation
 *
 * This simple calculator program shows how Mistode handles:
 * - Function declarations and definitions
 * - Variable names
 * - Comments
 * - Preprocessor directives (preserved)
 * - Standard library functions (preserved)
 */

#include <stdio.h>
#include <stdlib.h>

// Define constants
#define PI 3.14159265358979323846
#define MAX_OPERATIONS 100

// Structure to hold calculator state
struct Calculator {
  int operation_count;
  double last_result;
};

// Initialize calculator
void init_calculator(struct Calculator *calc) {
  calc->operation_count = 0;
  calc->last_result = 0.0;
}

// Calculate the area of a circle
double calculate_area(double radius) {
  double area = PI * radius * radius;
  return area;
}

// Calculate the volume of a cylinder
double calculate_volume(double radius, double height) {
  double base_area = calculate_area(radius);
  double volume = base_area * height;
  return volume;
}

// Perform addition
double add_numbers(struct Calculator *calc, double a, double b) {
  double result = a + b;
  calc->last_result = result;
  calc->operation_count++;
  return result;
}

// Perform multiplication
double multiply_numbers(struct Calculator *calc, double a, double b) {
  double result = a * b;
  calc->last_result = result;
  calc->operation_count++;
  return result;
}

// Print calculator statistics
void print_stats(struct Calculator *calc) {
  printf("Calculator Statistics:\n");
  printf("  Operations performed: %d\n", calc->operation_count);
  printf("  Last result: %.2f\n", calc->last_result);
}

int main() {
  struct Calculator calc;
  init_calculator(&calc);

  // Test geometric calculations
  double radius = 5.0;
  double height = 10.0;

  double area = calculate_area(radius);
  double volume = calculate_volume(radius, height);

  printf("Circle area (r=%.1f): %.2f\n", radius, area);
  printf("Cylinder volume (r=%.1f, h=%.1f): %.2f\n", radius, height, volume);

  // Test arithmetic operations
  double sum = add_numbers(&calc, 10.5, 20.3);
  double product = multiply_numbers(&calc, 3.14, 2.0);

  printf("\nArithmetic results:\n");
  printf("  10.5 + 20.3 = %.2f\n", sum);
  printf("  3.14 * 2.0 = %.2f\n", product);

  printf("\n");
  print_stats(&calc);

  return 0;
}

File: examples/calculator.py
"""
Example Python script demonstrating Mistode obfuscation

This simple calculator module shows how Mistode handles:
- Function definitions
- Parameter names
- Local variables
- Imported modules (preserved)
- Built-in functions (preserved)
"""

import math
from datetime import datetime


def calculate_area(radius):
    """Calculate the area of a circle given its radius"""
    pi_value = math.pi
    area = pi_value * radius**2
    return area


def calculate_volume(radius, height):
    """Calculate the volume of a cylinder"""
    base_area = calculate_area(radius)
    volume = base_area * height
    return volume


def format_result(value, unit="m²"):
    """Format a numerical result with units"""
    timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    result_str = f"[{timestamp}] Result: {value:.2f} {unit}"
    return result_str


class Calculator:
    """A simple calculator class"""

    def __init__(self, precision=2):
        self.precision = precision
        self.history = []

    def add(self, a, b):
        """Add two numbers"""
        result = a + b
        self.history.append(("add", a, b, result))
        return round(result, self.precision)

    def multiply(self, a, b):
        """Multiply two numbers"""
        result = a * b
        self.history.append(("multiply", a, b, result))
        return round(result, self.precision)

    def get_history(self):
        """Return calculation history"""
        return self.history


if __name__ == "__main__":
    # Test the functions
    r = 5.0
    h = 10.0

    area = calculate_area(r)
    volume = calculate_volume(r, h)

    print(format_result(area, "m²"))
    print(format_result(volume, "m³"))

    # Test the calculator class
    calc = Calculator(precision=3)
    result1 = calc.add(10.5, 20.3)
    result2 = calc.multiply(3.14, 2.0)

    print(f"Addition result: {result1}")
    print(f"Multiplication result: {result2}")
    print(f"History: {calc.get_history()}")
